<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">

<style type="text/css">
@font-face {
font-family: octicons-link;
src: url(data:font/woff;charset=utf-8;base64,d09GRgABAAAAAAZwABAAAAAACFQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABEU0lHAAAGaAAAAAgAAAAIAAAAAUdTVUIAAAZcAAAACgAAAAoAAQAAT1MvMgAAAyQAAABJAAAAYFYEU3RjbWFwAAADcAAAAEUAAACAAJThvmN2dCAAAATkAAAABAAAAAQAAAAAZnBnbQAAA7gAAACyAAABCUM+8IhnYXNwAAAGTAAAABAAAAAQABoAI2dseWYAAAFsAAABPAAAAZwcEq9taGVhZAAAAsgAAAA0AAAANgh4a91oaGVhAAADCAAAABoAAAAkCA8DRGhtdHgAAAL8AAAADAAAAAwGAACfbG9jYQAAAsAAAAAIAAAACABiATBtYXhwAAACqAAAABgAAAAgAA8ASm5hbWUAAAToAAABQgAAAlXu73sOcG9zdAAABiwAAAAeAAAAME3QpOBwcmVwAAAEbAAAAHYAAAB/aFGpk3jaTY6xa8JAGMW/O62BDi0tJLYQincXEypYIiGJjSgHniQ6umTsUEyLm5BV6NDBP8Tpts6F0v+k/0an2i+itHDw3v2+9+DBKTzsJNnWJNTgHEy4BgG3EMI9DCEDOGEXzDADU5hBKMIgNPZqoD3SilVaXZCER3/I7AtxEJLtzzuZfI+VVkprxTlXShWKb3TBecG11rwoNlmmn1P2WYcJczl32etSpKnziC7lQyWe1smVPy/Lt7Kc+0vWY/gAgIIEqAN9we0pwKXreiMasxvabDQMM4riO+qxM2ogwDGOZTXxwxDiycQIcoYFBLj5K3EIaSctAq2kTYiw+ymhce7vwM9jSqO8JyVd5RH9gyTt2+J/yUmYlIR0s04n6+7Vm1ozezUeLEaUjhaDSuXHwVRgvLJn1tQ7xiuVv/ocTRF42mNgZGBgYGbwZOBiAAFGJBIMAAizAFoAAABiAGIAznjaY2BkYGAA4in8zwXi+W2+MjCzMIDApSwvXzC97Z4Ig8N/BxYGZgcgl52BCSQKAA3jCV8CAABfAAAAAAQAAEB42mNgZGBg4f3vACQZQABIMjKgAmYAKEgBXgAAeNpjYGY6wTiBgZWBg2kmUxoDA4MPhGZMYzBi1AHygVLYQUCaawqDA4PChxhmh/8ODDEsvAwHgMKMIDnGL0x7gJQCAwMAJd4MFwAAAHjaY2BgYGaA4DAGRgYQkAHyGMF8NgYrIM3JIAGVYYDT+AEjAwuDFpBmA9KMDEwMCh9i/v8H8sH0/4dQc1iAmAkALaUKLgAAAHjaTY9LDsIgEIbtgqHUPpDi3gPoBVyRTmTddOmqTXThEXqrob2gQ1FjwpDvfwCBdmdXC5AVKFu3e5MfNFJ29KTQT48Ob9/lqYwOGZxeUelN2U2R6+cArgtCJpauW7UQBqnFkUsjAY/kOU1cP+DAgvxwn1chZDwUbd6CFimGXwzwF6tPbFIcjEl+vvmM/byA48e6tWrKArm4ZJlCbdsrxksL1AwWn/yBSJKpYbq8AXaaTb8AAHja28jAwOC00ZrBeQNDQOWO//sdBBgYGRiYWYAEELEwMTE4uzo5Zzo5b2BxdnFOcALxNjA6b2ByTswC8jYwg0VlNuoCTWAMqNzMzsoK1rEhNqByEyerg5PMJlYuVueETKcd/89uBpnpvIEVomeHLoMsAAe1Id4AAAAAAAB42oWQT07CQBTGv0JBhagk7HQzKxca2sJCE1hDt4QF+9JOS0nbaaYDCQfwCJ7Au3AHj+LO13FMmm6cl7785vven0kBjHCBhfpYuNa5Ph1c0e2Xu3jEvWG7UdPDLZ4N92nOm+EBXuAbHmIMSRMs+4aUEd4Nd3CHD8NdvOLTsA2GL8M9PODbcL+hD7C1xoaHeLJSEao0FEW14ckxC+TU8TxvsY6X0eLPmRhry2WVioLpkrbp84LLQPGI7c6sOiUzpWIWS5GzlSgUzzLBSikOPFTOXqly7rqx0Z1Q5BAIoZBSFihQYQOOBEdkCOgXTOHA07HAGjGWiIjaPZNW13/+lm6S9FT7rLHFJ6fQbkATOG1j2OFMucKJJsxIVfQORl+9Jyda6Sl1dUYhSCm1dyClfoeDve4qMYdLEbfqHf3O/AdDumsjAAB42mNgYoAAZQYjBmyAGYQZmdhL8zLdDEydARfoAqIAAAABAAMABwAKABMAB///AA8AAQAAAAAAAAAAAAAAAAABAAAAAA==) format('woff');
}
body {
-webkit-text-size-adjust: 100%;
text-size-adjust: 100%;
color: #333;
font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
font-size: 16px;
line-height: 1.6;
word-wrap: break-word;
}
a {
background-color: transparent;
}
a:active,
a:hover {
outline: 0;
}
strong {
font-weight: bold;
}
h1 {
font-size: 2em;
margin: 0.67em 0;
}
img {
border: 0;
}
hr {
box-sizing: content-box;
height: 0;
}
pre {
overflow: auto;
}
code,
kbd,
pre {
font-family: monospace, monospace;
font-size: 1em;
}
input {
color: inherit;
font: inherit;
margin: 0;
}
html input[disabled] {
cursor: default;
}
input {
line-height: normal;
}
input[type="checkbox"] {
box-sizing: border-box;
padding: 0;
}
table {
border-collapse: collapse;
border-spacing: 0;
}
td,
th {
padding: 0;
}
* {
box-sizing: border-box;
}
input {
font: 13px / 1.4 Helvetica, arial, nimbussansl, liberationsans, freesans, clean, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
a {
color: #4078c0;
text-decoration: none;
}
a:hover,
a:active {
text-decoration: underline;
}
hr {
height: 0;
margin: 15px 0;
overflow: hidden;
background: transparent;
border: 0;
border-bottom: 1px solid #ddd;
}
hr:before {
display: table;
content: "";
}
hr:after {
display: table;
clear: both;
content: "";
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 15px;
margin-bottom: 15px;
line-height: 1.1;
}
h1 {
font-size: 30px;
}
h2 {
font-size: 21px;
}
h3 {
font-size: 16px;
}
h4 {
font-size: 14px;
}
h5 {
font-size: 12px;
}
h6 {
font-size: 11px;
}
blockquote {
margin: 0;
}
ul,
ol {
padding: 0;
margin-top: 0;
margin-bottom: 0;
}
ol ol,
ul ol {
list-style-type: lower-roman;
}
ul ul ol,
ul ol ol,
ol ul ol,
ol ol ol {
list-style-type: lower-alpha;
}
dd {
margin-left: 0;
}
code {
font-family: Consolas, "Liberation Mono", Menlo, Courier, monospace;
font-size: 12px;
}
pre {
margin-top: 0;
margin-bottom: 0;
font: 12px Consolas, "Liberation Mono", Menlo, Courier, monospace;
}
.select::-ms-expand {
opacity: 0;
}
.octicon {
font: normal normal normal 16px/1 octicons-link;
display: inline-block;
text-decoration: none;
text-rendering: auto;
-webkit-font-smoothing: antialiased;
-moz-osx-font-smoothing: grayscale;
-webkit-user-select: none;
-moz-user-select: none;
-ms-user-select: none;
user-select: none;
}
.octicon-link:before {
content: '\f05c';
}
.markdown-body:before {
display: table;
content: "";
}
.markdown-body:after {
display: table;
clear: both;
content: "";
}
.markdown-body>*:first-child {
margin-top: 0 !important;
}
.markdown-body>*:last-child {
margin-bottom: 0 !important;
}
a:not([href]) {
color: inherit;
text-decoration: none;
}
.anchor {
display: inline-block;
padding-right: 2px;
margin-left: -18px;
}
.anchor:focus {
outline: none;
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 1em;
margin-bottom: 16px;
font-weight: bold;
line-height: 1.4;
}
h1 .octicon-link,
h2 .octicon-link,
h3 .octicon-link,
h4 .octicon-link,
h5 .octicon-link,
h6 .octicon-link {
color: #000;
vertical-align: middle;
visibility: hidden;
}
h1:hover .anchor,
h2:hover .anchor,
h3:hover .anchor,
h4:hover .anchor,
h5:hover .anchor,
h6:hover .anchor {
text-decoration: none;
}
h1:hover .anchor .octicon-link,
h2:hover .anchor .octicon-link,
h3:hover .anchor .octicon-link,
h4:hover .anchor .octicon-link,
h5:hover .anchor .octicon-link,
h6:hover .anchor .octicon-link {
visibility: visible;
}
h1 {
padding-bottom: 0.3em;
font-size: 2.25em;
line-height: 1.2;
border-bottom: 1px solid #eee;
}
h1 .anchor {
line-height: 1;
}
h2 {
padding-bottom: 0.3em;
font-size: 1.75em;
line-height: 1.225;
border-bottom: 1px solid #eee;
}
h2 .anchor {
line-height: 1;
}
h3 {
font-size: 1.5em;
line-height: 1.43;
}
h3 .anchor {
line-height: 1.2;
}
h4 {
font-size: 1.25em;
}
h4 .anchor {
line-height: 1.2;
}
h5 {
font-size: 1em;
}
h5 .anchor {
line-height: 1.1;
}
h6 {
font-size: 1em;
color: #777;
}
h6 .anchor {
line-height: 1.1;
}
p,
blockquote,
ul,
ol,
dl,
table,
pre {
margin-top: 0;
margin-bottom: 16px;
}
hr {
height: 4px;
padding: 0;
margin: 16px 0;
background-color: #e7e7e7;
border: 0 none;
}
ul,
ol {
padding-left: 2em;
}
ul ul,
ul ol,
ol ol,
ol ul {
margin-top: 0;
margin-bottom: 0;
}
li>p {
margin-top: 16px;
}
dl {
padding: 0;
}
dl dt {
padding: 0;
margin-top: 16px;
font-size: 1em;
font-style: italic;
font-weight: bold;
}
dl dd {
padding: 0 16px;
margin-bottom: 16px;
}
blockquote {
padding: 0 15px;
color: #777;
border-left: 4px solid #ddd;
}
blockquote>:first-child {
margin-top: 0;
}
blockquote>:last-child {
margin-bottom: 0;
}
table {
display: block;
width: 100%;
overflow: auto;
word-break: normal;
word-break: keep-all;
}
table th {
font-weight: bold;
}
table th,
table td {
padding: 6px 13px;
border: 1px solid #ddd;
}
table tr {
background-color: #fff;
border-top: 1px solid #ccc;
}
table tr:nth-child(2n) {
background-color: #f8f8f8;
}
img {
max-width: 100%;
box-sizing: content-box;
background-color: #fff;
}
code {
padding: 0;
padding-top: 0.2em;
padding-bottom: 0.2em;
margin: 0;
font-size: 85%;
background-color: rgba(0,0,0,0.04);
border-radius: 3px;
}
code:before,
code:after {
letter-spacing: -0.2em;
content: "\00a0";
}
pre>code {
padding: 0;
margin: 0;
font-size: 100%;
word-break: normal;
white-space: pre;
background: transparent;
border: 0;
}
.highlight {
margin-bottom: 16px;
}
.highlight pre,
pre {
padding: 16px;
overflow: auto;
font-size: 85%;
line-height: 1.45;
background-color: #f7f7f7;
border-radius: 3px;
}
.highlight pre {
margin-bottom: 0;
word-break: normal;
}
pre {
word-wrap: normal;
}
pre code {
display: inline;
max-width: initial;
padding: 0;
margin: 0;
overflow: initial;
line-height: inherit;
word-wrap: normal;
background-color: transparent;
border: 0;
}
pre code:before,
pre code:after {
content: normal;
}
kbd {
display: inline-block;
padding: 3px 5px;
font-size: 11px;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.pl-c {
color: #969896;
}
.pl-c1,
.pl-s .pl-v {
color: #0086b3;
}
.pl-e,
.pl-en {
color: #795da3;
}
.pl-s .pl-s1,
.pl-smi {
color: #333;
}
.pl-ent {
color: #63a35c;
}
.pl-k {
color: #a71d5d;
}
.pl-pds,
.pl-s,
.pl-s .pl-pse .pl-s1,
.pl-sr,
.pl-sr .pl-cce,
.pl-sr .pl-sra,
.pl-sr .pl-sre {
color: #183691;
}
.pl-v {
color: #ed6a43;
}
.pl-id {
color: #b52a1d;
}
.pl-ii {
background-color: #b52a1d;
color: #f8f8f8;
}
.pl-sr .pl-cce {
color: #63a35c;
font-weight: bold;
}
.pl-ml {
color: #693a17;
}
.pl-mh,
.pl-mh .pl-en,
.pl-ms {
color: #1d3e81;
font-weight: bold;
}
.pl-mq {
color: #008080;
}
.pl-mi {
color: #333;
font-style: italic;
}
.pl-mb {
color: #333;
font-weight: bold;
}
.pl-md {
background-color: #ffecec;
color: #bd2c00;
}
.pl-mi1 {
background-color: #eaffea;
color: #55a532;
}
.pl-mdr {
color: #795da3;
font-weight: bold;
}
.pl-mo {
color: #1d3e81;
}
kbd {
display: inline-block;
padding: 3px 5px;
font: 11px Consolas, "Liberation Mono", Menlo, Courier, monospace;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.task-list-item {
list-style-type: none;
}
.task-list-item+.task-list-item {
margin-top: 3px;
}
.task-list-item input {
margin: 0 0.35em 0.25em -1.6em;
vertical-align: middle;
}
:checked+.radio-label {
z-index: 1;
position: relative;
border-color: #4078c0;
}
.sourceLine {
display: inline-block;
}
code .kw { color: #000000; }
code .dt { color: #ed6a43; }
code .dv { color: #009999; }
code .bn { color: #009999; }
code .fl { color: #009999; }
code .ch { color: #009999; }
code .st { color: #183691; }
code .co { color: #969896; }
code .ot { color: #0086b3; }
code .al { color: #a61717; }
code .fu { color: #63a35c; }
code .er { color: #a61717; background-color: #e3d2d2; }
code .wa { color: #000000; }
code .cn { color: #008080; }
code .sc { color: #008080; }
code .vs { color: #183691; }
code .ss { color: #183691; }
code .im { color: #000000; }
code .va {color: #008080; }
code .cf { color: #000000; }
code .op { color: #000000; }
code .bu { color: #000000; }
code .ex { color: #000000; }
code .pp { color: #999999; }
code .at { color: #008080; }
code .do { color: #969896; }
code .an { color: #008080; }
code .cv { color: #008080; }
code .in { color: #008080; }
</style>
<style>
body {
box-sizing: border-box;
min-width: 200px;
max-width: 980px;
margin: 0 auto;
padding: 45px;
padding-top: 0px;
}
</style>


</head>

<body>

<h1 id="sourmash-phyloseq-object-creation">Sourmash phyloseq object
creation</h1>
<h2 id="gather">Gather</h2>
<p>The <code>sourmash gather</code> command was used to create 5 gather
csv files found in <code>examples/data/gather</code>. The parameters
used to generate these samples can be found in the file names.</p>
<p>The <code>podar-ref.zip</code> database is a small toy database for
easy testing.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="cf">for</span> sig <span class="kw">in</span> <span class="er">(</span><span class="fu">awk</span> <span class="at">-F,</span> <span class="st">&#39;NR&gt;1 {print $21}&#39;</span> examples/data/tiny_example.csv<span class="kw">);</span> <span class="cf">do</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>    <span class="ex">sourmash</span> gather examples/data/sigs/<span class="va">${sig}</span>.sig examples/data/podar-ref.zip <span class="at">-k</span> 31 <span class="at">--scaled</span> 1000 <span class="at">--dna</span> <span class="at">-o</span> examples/data/gather/<span class="va">${sig}</span>.k31-s1000-t3000.csv <span class="at">--threshold-bp</span> 3000</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<h2 id="tax">Tax</h2>
<p>The taxonomy file is located in <code>examples/data/</code> and is
<code>podar-red.tax.csv</code>.</p>
<h2 id="metadata">Metadata</h2>
<p>The metadata for the 5 example signatures used in this vingette is
from the <a href="https://github.com/waldronlab/curatedMetagenomicData">CuratedMetagenomicData
repository on GitHub</a></p>
<p>An example metadata file was generated with:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">head</span> <span class="at">-1</span> data/big_metadata.csv <span class="op">&gt;</span> tiny_example.csv</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> +2 data/big_metadata.csv <span class="kw">|</span> <span class="fu">shuf</span> <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n</span> 5 <span class="op">&gt;&gt;</span> tiny_example.csv</span></code></pre></div>
<h2 id="sourmash-gather-files-to-table">Sourmash gather files to
table</h2>
<p>The phyloseq object expects an n by m matrix. To achieve this across
a large number of samples, I wrote the <code>tables</code> plugin for
sourmash.</p>
<p>This plugin collects all the gather information and creates an OTU
object from our sourmash gather data.</p>
<p>An example of the tables plugin operation:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">sourmash</span> scripts gather_tables examples/data/gather/<span class="pp">*</span>  <span class="at">--format</span> dense <span class="at">-o</span> examples/data/dense-gather.k31-s1000-t3000.csv <span class="at">--column</span> f_unique_weighted <span class="at">--filter</span> 0</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">match_name</th>
<th align="right">ERR1136663</th>
<th align="right">ERR4087400</th>
<th align="right">ERR4562131</th>
<th align="right">SRR059888</th>
<th align="right">SRR9217435</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CP000139.1 Bacteroides vulgatus ATCC 8482, complete
genome</td>
<td align="right">0.0083015</td>
<td align="right">0.0055121</td>
<td align="right">0.0058230</td>
<td align="right">0.0000336</td>
<td align="right">0.0000579</td>
</tr>
<tr class="even">
<td align="left">AE015928.1 Bacteroides thetaiotaomicron VPI-5482,
complete genome</td>
<td align="right">0.0020327</td>
<td align="right">0.0003969</td>
<td align="right">0.0004230</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NZ_DS996397.1 Desulfovibrio piger ATCC 29098 Scfld442,
whole genome shotgun sequence</td>
<td align="right">0.0010764</td>
<td align="right">0.0000000</td>
<td align="right">0.0001549</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP001071.1 Akkermansia muciniphila ATCC BAA-835,
complete genome</td>
<td align="right">0.0005676</td>
<td align="right">0.0041199</td>
<td align="right">0.0000430</td>
<td align="right">0.0000098</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NZ_KE136524.1 Enterococcus faecalis V583
acyDH-supercont2.1, whole genome shotgun sequence</td>
<td align="right">0.0000236</td>
<td align="right">0.0000693</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP001472.1 Acidobacterium capsulatum ATCC 51196,
complete genome</td>
<td align="right">0.0001166</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">AE009951.2 Fusobacterium nucleatum subsp. nucleatum
ATCC 25586, complete genome</td>
<td align="right">0.0002815</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0032173</td>
<td align="right">0.0000565</td>
</tr>
<tr class="even">
<td align="left">CP001013.1 Leptothrix cholodnii SP-6, complete
genome</td>
<td align="right">0.0000094</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001588</td>
<td align="right">0.0001924</td>
</tr>
<tr class="odd">
<td align="left">NZ_JGWU01000001.1 Bordetella bronchiseptica D989
ctg7180000008197, whole genome shotgun sequence</td>
<td align="right">0.0000000</td>
<td align="right">0.0000378</td>
<td align="right">0.0000000</td>
<td align="right">0.0002794</td>
<td align="right">0.0003983</td>
</tr>
<tr class="even">
<td align="left">NZ_KQ961402.1 Zymomonas mobilis strain ATCC 31823
Scaffold1, whole genome shotgun sequence</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003728</td>
<td align="right">0.0000000</td>
<td align="right">0.0000293</td>
</tr>
<tr class="odd">
<td align="left">AP009380.1 Porphyromonas gingivalis ATCC 33277 DNA,
complete genome</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000574</td>
<td align="right">0.0002765</td>
<td align="right">0.0000086</td>
</tr>
<tr class="even">
<td align="left">NC_007951.1 Burkholderia xenovorans LB400 chromosome 1,
complete sequence</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000258</td>
<td align="right">0.0000000</td>
<td align="right">0.0000043</td>
</tr>
<tr class="odd">
<td align="left">AE017226.1 Treponema denticola ATCC 35405, complete
genome</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000726</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP000850.1 Salinispora arenicola CNS-205, complete
genome</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003465</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">AE006470.1 Chlorobium tepidum TLS, complete genome</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000754</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">AE017221.1 Thermus thermophilus HB27, complete
genome</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000034</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NC_011663.1 Shewanella baltica OS223, complete
genome</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001623</td>
</tr>
</tbody>
</table>
<h2 id="r-script-to-create-sourmash-phyloseq-object">R script to create
sourmash phyloseq object</h2>
<h3 id="load-necessary-libraries">Load necessary libraries</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">library</span>(stringr)</span></code></pre></div>
<h3 id="load-sourmash-otu-and-tax-data">Load sourmash OTU and tax
data</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;data/dense-gather.k31-s1000-t3000.csv&#39;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>tax <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;data/podar-ref.tax.csv&#39;</span>)</span></code></pre></div>
<h3 id="manipulate-the-taxonomy-strings-in-both-dataframes-to-have-matching-row-names">Manipulate
the taxonomy strings in both dataframes to have matching row names</h3>
<p>View the sourmash OTU data</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">str</span>(df)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    17 obs. of  6 variables:
##  $ match_name: chr  &quot;CP000139.1 Bacteroides vulgatus ATCC 8482, complete genome&quot; &quot;AE015928.1 Bacteroides thetaiotaomicron VPI-5482, complete genome&quot; &quot;NZ_DS996397.1 Desulfovibrio piger ATCC 29098 Scfld442, whole genome shotgun sequence&quot; &quot;CP001071.1 Akkermansia muciniphila ATCC BAA-835, complete genome&quot; ...
##  $ ERR1136663: num  8.30e-03 2.03e-03 1.08e-03 5.68e-04 2.36e-05 ...
##  $ ERR4087400: num  5.51e-03 3.97e-04 0.00 4.12e-03 6.93e-05 ...
##  $ ERR4562131: num  0.005823 0.000423 0.000155 0.000043 0 ...
##  $ SRR059888 : num  3.36e-05 0.00 0.00 9.81e-06 0.00 ...
##  $ SRR9217435: num  5.79e-05 0.00 0.00 0.00 0.00 ...</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>df[,<span class="dv">1</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;CP000139.1 Bacteroides vulgatus ATCC 8482, complete genome&quot;                                      
##  [2] &quot;AE015928.1 Bacteroides thetaiotaomicron VPI-5482, complete genome&quot;                               
##  [3] &quot;NZ_DS996397.1 Desulfovibrio piger ATCC 29098 Scfld442, whole genome shotgun sequence&quot;            
##  [4] &quot;CP001071.1 Akkermansia muciniphila ATCC BAA-835, complete genome&quot;                                
##  [5] &quot;NZ_KE136524.1 Enterococcus faecalis V583 acyDH-supercont2.1, whole genome shotgun sequence&quot;      
##  [6] &quot;CP001472.1 Acidobacterium capsulatum ATCC 51196, complete genome&quot;                                
##  [7] &quot;AE009951.2 Fusobacterium nucleatum subsp. nucleatum ATCC 25586, complete genome&quot;                 
##  [8] &quot;CP001013.1 Leptothrix cholodnii SP-6, complete genome&quot;                                           
##  [9] &quot;NZ_JGWU01000001.1 Bordetella bronchiseptica D989 ctg7180000008197, whole genome shotgun sequence&quot;
## [10] &quot;NZ_KQ961402.1 Zymomonas mobilis strain ATCC 31823 Scaffold1, whole genome shotgun sequence&quot;      
## [11] &quot;AP009380.1 Porphyromonas gingivalis ATCC 33277 DNA, complete genome&quot;                             
## [12] &quot;NC_007951.1 Burkholderia xenovorans LB400 chromosome 1, complete sequence&quot;                       
## [13] &quot;AE017226.1 Treponema denticola ATCC 35405, complete genome&quot;                                      
## [14] &quot;CP000850.1 Salinispora arenicola CNS-205, complete genome&quot;                                       
## [15] &quot;AE006470.1 Chlorobium tepidum TLS, complete genome&quot;                                              
## [16] &quot;AE017221.1 Thermus thermophilus HB27, complete genome&quot;                                           
## [17] &quot;NC_011663.1 Shewanella baltica OS223, complete genome&quot;</code></pre>
<p>Remove the version id and the superfluous words</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">str_view</span>(<span class="fu">head</span>(df[,<span class="dv">1</span>]), <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.</span><span class="sc">\\</span><span class="st">d+&quot;</span>)</span></code></pre></div>
<pre><code>## [1] │ CP000139&lt;.1&gt; Bacteroides vulgatus ATCC 8482, complete genome
## [2] │ AE015928&lt;.1&gt; Bacteroides thetaiotaomicron VPI-5482, complete genome
## [3] │ NZ_DS996397&lt;.1&gt; Desulfovibrio piger ATCC 29098 Scfld442, whole genome shotgun sequence
## [4] │ CP001071&lt;.1&gt; Akkermansia muciniphila ATCC BAA-835, complete genome
## [5] │ NZ_KE136524&lt;.1&gt; Enterococcus faecalis V583 acyDH-supercont2&lt;.1&gt;, whole genome shotgun sequence
## [6] │ CP001472&lt;.1&gt; Acidobacterium capsulatum ATCC 51196, complete genome</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">word</span>(df[, <span class="dv">1</span>], <span class="dv">1</span>, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##  [1] &quot;CP000139.1 Bacteroides vulgatus&quot;            
##  [2] &quot;AE015928.1 Bacteroides thetaiotaomicron&quot;    
##  [3] &quot;NZ_DS996397.1 Desulfovibrio piger&quot;          
##  [4] &quot;CP001071.1 Akkermansia muciniphila&quot;         
##  [5] &quot;NZ_KE136524.1 Enterococcus faecalis&quot;        
##  [6] &quot;CP001472.1 Acidobacterium capsulatum&quot;       
##  [7] &quot;AE009951.2 Fusobacterium nucleatum&quot;         
##  [8] &quot;CP001013.1 Leptothrix cholodnii&quot;            
##  [9] &quot;NZ_JGWU01000001.1 Bordetella bronchiseptica&quot;
## [10] &quot;NZ_KQ961402.1 Zymomonas mobilis&quot;            
## [11] &quot;AP009380.1 Porphyromonas gingivalis&quot;        
## [12] &quot;NC_007951.1 Burkholderia xenovorans&quot;        
## [13] &quot;AE017226.1 Treponema denticola&quot;             
## [14] &quot;CP000850.1 Salinispora arenicola&quot;           
## [15] &quot;AE006470.1 Chlorobium tepidum&quot;              
## [16] &quot;AE017221.1 Thermus thermophilus&quot;            
## [17] &quot;NC_011663.1 Shewanella baltica&quot;</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>df[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">word</span>(df[,<span class="dv">1</span>], <span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>df[,<span class="dv">1</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;CP000139.1 Bacteroides vulgatus&quot;            
##  [2] &quot;AE015928.1 Bacteroides thetaiotaomicron&quot;    
##  [3] &quot;NZ_DS996397.1 Desulfovibrio piger&quot;          
##  [4] &quot;CP001071.1 Akkermansia muciniphila&quot;         
##  [5] &quot;NZ_KE136524.1 Enterococcus faecalis&quot;        
##  [6] &quot;CP001472.1 Acidobacterium capsulatum&quot;       
##  [7] &quot;AE009951.2 Fusobacterium nucleatum&quot;         
##  [8] &quot;CP001013.1 Leptothrix cholodnii&quot;            
##  [9] &quot;NZ_JGWU01000001.1 Bordetella bronchiseptica&quot;
## [10] &quot;NZ_KQ961402.1 Zymomonas mobilis&quot;            
## [11] &quot;AP009380.1 Porphyromonas gingivalis&quot;        
## [12] &quot;NC_007951.1 Burkholderia xenovorans&quot;        
## [13] &quot;AE017226.1 Treponema denticola&quot;             
## [14] &quot;CP000850.1 Salinispora arenicola&quot;           
## [15] &quot;AE006470.1 Chlorobium tepidum&quot;              
## [16] &quot;AE017221.1 Thermus thermophilus&quot;            
## [17] &quot;NC_011663.1 Shewanella baltica&quot;</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>df[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(df[,<span class="dv">1</span>], <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.</span><span class="sc">\\</span><span class="st">d+&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>df[,<span class="dv">1</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;CP000139 Bacteroides vulgatus&quot;            
##  [2] &quot;AE015928 Bacteroides thetaiotaomicron&quot;    
##  [3] &quot;NZ_DS996397 Desulfovibrio piger&quot;          
##  [4] &quot;CP001071 Akkermansia muciniphila&quot;         
##  [5] &quot;NZ_KE136524 Enterococcus faecalis&quot;        
##  [6] &quot;CP001472 Acidobacterium capsulatum&quot;       
##  [7] &quot;AE009951 Fusobacterium nucleatum&quot;         
##  [8] &quot;CP001013 Leptothrix cholodnii&quot;            
##  [9] &quot;NZ_JGWU01000001 Bordetella bronchiseptica&quot;
## [10] &quot;NZ_KQ961402 Zymomonas mobilis&quot;            
## [11] &quot;AP009380 Porphyromonas gingivalis&quot;        
## [12] &quot;NC_007951 Burkholderia xenovorans&quot;        
## [13] &quot;AE017226 Treponema denticola&quot;             
## [14] &quot;CP000850 Salinispora arenicola&quot;           
## [15] &quot;AE006470 Chlorobium tepidum&quot;              
## [16] &quot;AE017221 Thermus thermophilus&quot;            
## [17] &quot;NC_011663 Shewanella baltica&quot;</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(df, <span class="at">format =</span> <span class="st">&#39;markdown&#39;</span>)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">match_name</th>
<th align="right">ERR1136663</th>
<th align="right">ERR4087400</th>
<th align="right">ERR4562131</th>
<th align="right">SRR059888</th>
<th align="right">SRR9217435</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CP000139 Bacteroides vulgatus</td>
<td align="right">0.0083015</td>
<td align="right">0.0055121</td>
<td align="right">0.0058230</td>
<td align="right">0.0000336</td>
<td align="right">0.0000579</td>
</tr>
<tr class="even">
<td align="left">AE015928 Bacteroides thetaiotaomicron</td>
<td align="right">0.0020327</td>
<td align="right">0.0003969</td>
<td align="right">0.0004230</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NZ_DS996397 Desulfovibrio piger</td>
<td align="right">0.0010764</td>
<td align="right">0.0000000</td>
<td align="right">0.0001549</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP001071 Akkermansia muciniphila</td>
<td align="right">0.0005676</td>
<td align="right">0.0041199</td>
<td align="right">0.0000430</td>
<td align="right">0.0000098</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NZ_KE136524 Enterococcus faecalis</td>
<td align="right">0.0000236</td>
<td align="right">0.0000693</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP001472 Acidobacterium capsulatum</td>
<td align="right">0.0001166</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">AE009951 Fusobacterium nucleatum</td>
<td align="right">0.0002815</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0032173</td>
<td align="right">0.0000565</td>
</tr>
<tr class="even">
<td align="left">CP001013 Leptothrix cholodnii</td>
<td align="right">0.0000094</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001588</td>
<td align="right">0.0001924</td>
</tr>
<tr class="odd">
<td align="left">NZ_JGWU01000001 Bordetella bronchiseptica</td>
<td align="right">0.0000000</td>
<td align="right">0.0000378</td>
<td align="right">0.0000000</td>
<td align="right">0.0002794</td>
<td align="right">0.0003983</td>
</tr>
<tr class="even">
<td align="left">NZ_KQ961402 Zymomonas mobilis</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003728</td>
<td align="right">0.0000000</td>
<td align="right">0.0000293</td>
</tr>
<tr class="odd">
<td align="left">AP009380 Porphyromonas gingivalis</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000574</td>
<td align="right">0.0002765</td>
<td align="right">0.0000086</td>
</tr>
<tr class="even">
<td align="left">NC_007951 Burkholderia xenovorans</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000258</td>
<td align="right">0.0000000</td>
<td align="right">0.0000043</td>
</tr>
<tr class="odd">
<td align="left">AE017226 Treponema denticola</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000726</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP000850 Salinispora arenicola</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003465</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">AE006470 Chlorobium tepidum</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000754</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">AE017221 Thermus thermophilus</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000034</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NC_011663 Shewanella baltica</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001623</td>
</tr>
</tbody>
</table>
<p>Those cleaned names need to be moved to row names (aka row ids)</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">rownames</span>(df) <span class="ot">&lt;-</span> df[,<span class="dv">1</span>]</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">rownames</span>(df)</span></code></pre></div>
<pre><code>##  [1] &quot;CP000139 Bacteroides vulgatus&quot;            
##  [2] &quot;AE015928 Bacteroides thetaiotaomicron&quot;    
##  [3] &quot;NZ_DS996397 Desulfovibrio piger&quot;          
##  [4] &quot;CP001071 Akkermansia muciniphila&quot;         
##  [5] &quot;NZ_KE136524 Enterococcus faecalis&quot;        
##  [6] &quot;CP001472 Acidobacterium capsulatum&quot;       
##  [7] &quot;AE009951 Fusobacterium nucleatum&quot;         
##  [8] &quot;CP001013 Leptothrix cholodnii&quot;            
##  [9] &quot;NZ_JGWU01000001 Bordetella bronchiseptica&quot;
## [10] &quot;NZ_KQ961402 Zymomonas mobilis&quot;            
## [11] &quot;AP009380 Porphyromonas gingivalis&quot;        
## [12] &quot;NC_007951 Burkholderia xenovorans&quot;        
## [13] &quot;AE017226 Treponema denticola&quot;             
## [14] &quot;CP000850 Salinispora arenicola&quot;           
## [15] &quot;AE006470 Chlorobium tepidum&quot;              
## [16] &quot;AE017221 Thermus thermophilus&quot;            
## [17] &quot;NC_011663 Shewanella baltica&quot;</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> df[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(df, <span class="at">format =</span> <span class="st">&#39;markdown&#39;</span>)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">ERR1136663</th>
<th align="right">ERR4087400</th>
<th align="right">ERR4562131</th>
<th align="right">SRR059888</th>
<th align="right">SRR9217435</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CP000139 Bacteroides vulgatus</td>
<td align="right">0.0083015</td>
<td align="right">0.0055121</td>
<td align="right">0.0058230</td>
<td align="right">0.0000336</td>
<td align="right">0.0000579</td>
</tr>
<tr class="even">
<td align="left">AE015928 Bacteroides thetaiotaomicron</td>
<td align="right">0.0020327</td>
<td align="right">0.0003969</td>
<td align="right">0.0004230</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NZ_DS996397 Desulfovibrio piger</td>
<td align="right">0.0010764</td>
<td align="right">0.0000000</td>
<td align="right">0.0001549</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP001071 Akkermansia muciniphila</td>
<td align="right">0.0005676</td>
<td align="right">0.0041199</td>
<td align="right">0.0000430</td>
<td align="right">0.0000098</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NZ_KE136524 Enterococcus faecalis</td>
<td align="right">0.0000236</td>
<td align="right">0.0000693</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP001472 Acidobacterium capsulatum</td>
<td align="right">0.0001166</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">AE009951 Fusobacterium nucleatum</td>
<td align="right">0.0002815</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0032173</td>
<td align="right">0.0000565</td>
</tr>
<tr class="even">
<td align="left">CP001013 Leptothrix cholodnii</td>
<td align="right">0.0000094</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001588</td>
<td align="right">0.0001924</td>
</tr>
<tr class="odd">
<td align="left">NZ_JGWU01000001 Bordetella bronchiseptica</td>
<td align="right">0.0000000</td>
<td align="right">0.0000378</td>
<td align="right">0.0000000</td>
<td align="right">0.0002794</td>
<td align="right">0.0003983</td>
</tr>
<tr class="even">
<td align="left">NZ_KQ961402 Zymomonas mobilis</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003728</td>
<td align="right">0.0000000</td>
<td align="right">0.0000293</td>
</tr>
<tr class="odd">
<td align="left">AP009380 Porphyromonas gingivalis</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000574</td>
<td align="right">0.0002765</td>
<td align="right">0.0000086</td>
</tr>
<tr class="even">
<td align="left">NC_007951 Burkholderia xenovorans</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000258</td>
<td align="right">0.0000000</td>
<td align="right">0.0000043</td>
</tr>
<tr class="odd">
<td align="left">AE017226 Treponema denticola</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000726</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">CP000850 Salinispora arenicola</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003465</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">AE006470 Chlorobium tepidum</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000754</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">AE017221 Thermus thermophilus</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000034</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NC_011663 Shewanella baltica</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001623</td>
</tr>
</tbody>
</table>
<p>View the tax information</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">str</span>(tax)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    64 obs. of  10 variables:
##  $ accession   : chr  &quot;AE000782&quot; &quot;NC_000909&quot; &quot;NC_003272&quot; &quot;AE009441&quot; ...
##  $ taxid       : int  224325 243232 103690 178306 186497 190304 188937 190192 246200 194439 ...
##  $ superkingdom: chr  &quot;Archaea&quot; &quot;Archaea&quot; &quot;Bacteria&quot; &quot;Archaea&quot; ...
##  $ phylum      : chr  &quot;Euryarchaeota&quot; &quot;Euryarchaeota&quot; &quot;Cyanobacteria&quot; &quot;Crenarchaeota&quot; ...
##  $ class       : chr  &quot;Archaeoglobi&quot; &quot;Methanococci&quot; &quot;&quot; &quot;Thermoprotei&quot; ...
##  $ order       : chr  &quot;Archaeoglobales&quot; &quot;Methanococcales&quot; &quot;Nostocales&quot; &quot;Thermoproteales&quot; ...
##  $ family      : chr  &quot;Archaeoglobaceae&quot; &quot;Methanocaldococcaceae&quot; &quot;Nostocaceae&quot; &quot;Thermoproteaceae&quot; ...
##  $ genus       : chr  &quot;Archaeoglobus&quot; &quot;Methanocaldococcus&quot; &quot;Nostoc&quot; &quot;Pyrobaculum&quot; ...
##  $ species     : chr  &quot;Archaeoglobus fulgidus&quot; &quot;Methanocaldococcus jannaschii&quot; &quot;Nostoc sp. PCC 7120&quot; &quot;Pyrobaculum aerophilum&quot; ...
##  $ strain      : chr  &quot;Archaeoglobus fulgidus DSM 4304&quot; &quot;Methanocaldococcus jannaschii DSM 2661&quot; &quot;&quot; &quot;Pyrobaculum aerophilum str. IM2&quot; ...</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>tax[,<span class="dv">1</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;AE000782&quot;        &quot;NC_000909&quot;       &quot;NC_003272&quot;       &quot;AE009441&quot;       
##  [5] &quot;AE009950&quot;        &quot;AE009951&quot;        &quot;AE010299&quot;        &quot;AE009439&quot;       
##  [9] &quot;NC_003911&quot;       &quot;AE006470&quot;        &quot;AE015928&quot;        &quot;AL954747&quot;       
## [13] &quot;BX119912&quot;        &quot;BX571656&quot;        &quot;AE017180&quot;        &quot;AE017226&quot;       
## [17] &quot;BX950229&quot;        &quot;AE017221&quot;        &quot;BA000001&quot;        &quot;BA000023&quot;       
## [21] &quot;NC_007951&quot;       &quot;CP000492&quot;        &quot;NC_008751&quot;       &quot;CP000568&quot;       
## [25] &quot;CP000561&quot;        &quot;CP000609&quot;        &quot;CP000607&quot;        &quot;CP000660&quot;       
## [29] &quot;CP000667&quot;        &quot;CP000679&quot;        &quot;CP000702&quot;        &quot;CP000139&quot;       
## [33] &quot;NC_009665&quot;       &quot;CP000816&quot;        &quot;CP000850&quot;        &quot;CP000909&quot;       
## [37] &quot;CP000924&quot;        &quot;CP000969&quot;        &quot;CP001013&quot;        &quot;CP001071&quot;       
## [41] &quot;AP009380&quot;        &quot;NC_010730&quot;       &quot;CP001097&quot;        &quot;CP001110&quot;       
## [45] &quot;CP001130&quot;        &quot;NZ_CH959311&quot;     &quot;NZ_CH959317&quot;     &quot;CP001251&quot;       
## [49] &quot;NC_011663&quot;       &quot;CP000916&quot;        &quot;NZ_DS996397&quot;     &quot;CP001230&quot;       
## [53] &quot;CP001472&quot;        &quot;AP009153&quot;        &quot;CP001941&quot;        &quot;NC_013968&quot;      
## [57] &quot;NZ_KE136524&quot;     &quot;NZ_KQ961402&quot;     &quot;NZ_CP015081&quot;     &quot;NZ_ABZS01000228&quot;
## [61] &quot;NZ_JGWU01000001&quot; &quot;NZ_FWDH01000003&quot; &quot;NC_009972&quot;       &quot;NC_005213&quot;</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>tax[,<span class="dv">9</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;Archaeoglobus fulgidus&quot;              
##  [2] &quot;Methanocaldococcus jannaschii&quot;       
##  [3] &quot;Nostoc sp. PCC 7120&quot;                 
##  [4] &quot;Pyrobaculum aerophilum&quot;              
##  [5] &quot;Pyrococcus furiosus&quot;                 
##  [6] &quot;Fusobacterium nucleatum&quot;             
##  [7] &quot;Methanosarcina acetivorans&quot;          
##  [8] &quot;Methanopyrus kandleri&quot;               
##  [9] &quot;Ruegeria pomeroyi&quot;                   
## [10] &quot;Chlorobaculum tepidum&quot;               
## [11] &quot;Bacteroides thetaiotaomicron&quot;        
## [12] &quot;Nitrosomonas europaea&quot;               
## [13] &quot;Rhodopirellula baltica&quot;              
## [14] &quot;Wolinella succinogenes&quot;              
## [15] &quot;Geobacter sulfurreducens&quot;            
## [16] &quot;Treponema denticola&quot;                 
## [17] &quot;Methanococcus maripaludis&quot;           
## [18] &quot;Thermus thermophilus&quot;                
## [19] &quot;Pyrococcus horikoshii&quot;               
## [20] &quot;Sulfolobus tokodaii&quot;                 
## [21] &quot;Paraburkholderia xenovorans&quot;         
## [22] &quot;Chlorobium phaeobacteroides&quot;         
## [23] &quot;Desulfovibrio vulgaris&quot;              
## [24] &quot;Ruminiclostridium thermocellum&quot;      
## [25] &quot;Pyrobaculum calidifontis&quot;            
## [26] &quot;Methanococcus maripaludis&quot;           
## [27] &quot;Chlorobium phaeovibrioides&quot;          
## [28] &quot;Pyrobaculum arsenaticum&quot;             
## [29] &quot;Salinispora tropica&quot;                 
## [30] &quot;Caldicellulosiruptor saccharolyticus&quot;
## [31] &quot;Thermotoga petrophila&quot;               
## [32] &quot;Bacteroides vulgatus&quot;                
## [33] &quot;Shewanella baltica&quot;                  
## [34] &quot;Ignicoccus hospitalis&quot;               
## [35] &quot;Salinispora arenicola&quot;               
## [36] &quot;Chloroflexus aurantiacus&quot;            
## [37] &quot;Thermoanaerobacter pseudethanolicus&quot; 
## [38] &quot;Thermotoga sp. RQ2&quot;                  
## [39] &quot;Leptothrix cholodnii&quot;                
## [40] &quot;Akkermansia muciniphila&quot;             
## [41] &quot;Porphyromonas gingivalis&quot;            
## [42] &quot;Sulfurihydrogenibium sp. YO3AOP1&quot;    
## [43] &quot;Chlorobium limicola&quot;                 
## [44] &quot;Pelodictyon phaeoclathratiforme&quot;     
## [45] &quot;Hydrogenobaculum sp. Y04AAS1&quot;        
## [46] &quot;Sulfitobacter sp. EE-36&quot;             
## [47] &quot;Sulfitobacter sp. NAS-14.1&quot;          
## [48] &quot;Dictyoglomus turgidum&quot;               
## [49] &quot;Shewanella baltica&quot;                  
## [50] &quot;Thermotoga neapolitana&quot;              
## [51] &quot;Desulfovibrio piger&quot;                 
## [52] &quot;Persephonella marina&quot;                
## [53] &quot;Acidobacterium capsulatum&quot;           
## [54] &quot;Gemmatimonas aurantiaca&quot;             
## [55] &quot;Aciduliprofundum boonei&quot;             
## [56] &quot;Haloferax volcanii&quot;                  
## [57] &quot;Enterococcus faecalis&quot;               
## [58] &quot;Zymomonas mobilis&quot;                   
## [59] &quot;Deinococcus radiodurans&quot;             
## [60] &quot;Sulfurihydrogenibium yellowstonense&quot; 
## [61] &quot;Bordetella bronchiseptica&quot;           
## [62] &quot;Caldicellulosiruptor bescii&quot;         
## [63] &quot;Herpetosiphon aurantiacus&quot;           
## [64] &quot;Nanoarchaeum equitans&quot;</code></pre>
<p>We do not need the TaxID. We do need to match the row names of the
sourmash OTU, though. Therefore, we need to collapse the accession and
species name into a single string and move that string to the row
names</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>tax <span class="ot">&lt;-</span> tax[,<span class="sc">-</span><span class="dv">2</span>]</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="fu">str_view</span>(<span class="fu">head</span>(tax<span class="sc">$</span>accession), <span class="st">&quot;.*&quot;</span>)</span></code></pre></div>
<pre><code>## [1] │ &lt;AE000782&gt;&lt;&gt;
## [2] │ &lt;NC_000909&gt;&lt;&gt;
## [3] │ &lt;NC_003272&gt;&lt;&gt;
## [4] │ &lt;AE009441&gt;&lt;&gt;
## [5] │ &lt;AE009950&gt;&lt;&gt;
## [6] │ &lt;AE009951&gt;&lt;&gt;</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">rownames</span>(tax) <span class="ot">&lt;-</span> <span class="fu">str_glue_data</span>(tax, <span class="st">&quot;{accession} {species}&quot;</span>)</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>tax <span class="ot">&lt;-</span> tax[,<span class="sc">-</span><span class="dv">1</span>]</span></code></pre></div>
<h2 id="make-the-phyloseq-object">Make the phyloseq object</h2>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>OTU <span class="ot">=</span> <span class="fu">otu_table</span>(df, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>TAX <span class="ot">=</span> <span class="fu">tax_table</span>(<span class="fu">as.matrix</span>(tax))</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">phyloseq</span>(OTU, TAX)</span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>physeq</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 15 taxa and 5 samples ]
## tax_table()   Taxonomy Table:    [ 15 taxa by 8 taxonomic ranks ]</code></pre>
<h2 id="create-a-metadata-dataframe-to-use-with-the-sourmash-phyloseq-object">Create
a metadata dataframe to use with the sourmash phyloseq object</h2>
<p>Read in the csv file that contains your sample metadata</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>all_metadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/tiny_example.csv&quot;</span>)</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(all_metadata)</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a><span class="fu">str</span>(metadata)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    5 obs. of  142 variables:
## Formal class &#39;sample_data&#39; [package &quot;phyloseq&quot;] with 4 slots
##   ..@ .Data    :List of 142
##   .. ..$ : chr  &quot;ZeeviD_2015&quot; &quot;MetaCardis_2020_a&quot; &quot;MetaCardis_2020_a&quot; &quot;GhensiP_2019&quot; ...
##   .. ..$ : chr  &quot;PNP_Main_553&quot; &quot;M0x20MCx1812&quot; &quot;M0x20MCx1360&quot; &quot;SP_101SPI_T016&quot; ...
##   .. ..$ : chr  &quot;PNP_Main_553&quot; &quot;M0x20MCx1812&quot; &quot;M0x20MCx1360&quot; &quot;sub_101&quot; ...
##   .. ..$ : chr  &quot;stool&quot; &quot;stool&quot; &quot;stool&quot; &quot;oralcavity&quot; ...
##   .. ..$ : chr  &quot;no&quot; &quot;no&quot; &quot;yes&quot; NA ...
##   .. ..$ : chr  &quot;control&quot; &quot;T2D&quot; &quot;IGT&quot; &quot;control&quot; ...
##   .. ..$ : chr  &quot;healthy&quot; &quot;T2D&quot; &quot;IGT;MS&quot; &quot;healthy&quot; ...
##   .. ..$ : int  51 63 51 NA 27
##   .. ..$ : chr  &quot;adult&quot; &quot;adult&quot; &quot;adult&quot; &quot;adult&quot; ...
##   .. ..$ : chr  &quot;female&quot; &quot;male&quot; &quot;female&quot; NA ...
##   .. ..$ : chr  &quot;ISR&quot; &quot;DEU&quot; &quot;DEU&quot; &quot;ITA&quot; ...
##   .. ..$ : chr  NA &quot;Leipzig&quot; &quot;Leipzig&quot; NA ...
##   .. ..$ : chr  &quot;no&quot; &quot;no&quot; &quot;no&quot; &quot;no&quot; ...
##   .. ..$ : chr  &quot;IlluminaHiSeq&quot; &quot;IonProton&quot; &quot;IonProton&quot; &quot;IlluminaHiSeq&quot; ...
##   .. ..$ : chr  NA NA NA NA ...
##   .. ..$ : int  26590418 34880489 34880489 33127901 22699609
##   .. ..$ : int  28756422 15374062 20888458 20061461 94436101
##   .. ..$ : num  2.52e+09 2.33e+09 3.20e+09 1.99e+09 8.71e+09
##   .. ..$ : int  20 NA NA 75 60
##   .. ..$ : int  80 NA NA 100 101
##   .. ..$ : chr  &quot;ERR1136663&quot; &quot;ERR4562131&quot; &quot;ERR4087400&quot; &quot;SRR9217435&quot; ...
##   .. ..$ : num  NA NA 6.78 NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : chr  &quot;no&quot; &quot;antidiab;metformin;dppiv&quot; &quot;antihta;ca2_cbl&quot; NA ...
##   .. ..$ : chr  &quot;Jacob_Wirbel;Paolo_Manghi&quot; &quot;Paolo_Manghi&quot; &quot;Paolo_Manghi&quot; &quot;Paolo_Manghi&quot; ...
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : chr  &quot;no&quot; NA NA NA ...
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : num  23.7 27.7 50.1 NA 23
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : chr  NA &quot;yes&quot; &quot;no&quot; &quot;no&quot; ...
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : chr  &quot;rectal_swab&quot; NA NA &quot;subgingival_plaque&quot; ...
##   .. ..$ : chr  &quot;no_immuno_suppressive;no_T2D;no_T1D;no_related_treatments;no_psychiatric_diseases;no_gastro_intestinal_disorder;non_celiac&quot; NA NA NA ...
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : num  NA NA 210 NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : num  NA NA 240 NA NA
##   .. ..$ : num  NA NA 5.73 NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : chr  NA NA NA &quot;yes&quot; ...
##   .. ..$ : chr  NA NA NA &quot;implant&quot; ...
##   .. ..$ : chr  NA NA NA &quot;yes&quot; ...
##   .. ..$ : int  NA NA NA 4 NA
##   .. ..$ : int  NA NA NA 3 NA
##   .. ..$ : int  NA NA NA 4 NA
##   .. ..$ : int  NA NA NA 3 NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. ..$ : logi  NA NA NA NA NA
##   .. .. [list output truncated]
##   ..@ names    : chr  &quot;study_name&quot; &quot;sample_id&quot; &quot;subject_id&quot; &quot;body_site&quot; ...
##   ..@ row.names: chr  &quot;sa1&quot; &quot;sa2&quot; &quot;sa3&quot; &quot;sa4&quot; ...
##   ..@ .S3Class : chr &quot;data.frame&quot;</code></pre>
<p>This file contained many columns we do not need. Lets remove
those.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata[, <span class="fu">colSums</span>(<span class="sc">!</span><span class="fu">is.na</span>(metadata)) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="fu">str</span>(metadata)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    5 obs. of  41 variables:
## Formal class &#39;sample_data&#39; [package &quot;phyloseq&quot;] with 4 slots
##   ..@ .Data    :List of 41
##   .. ..$ : chr  &quot;ZeeviD_2015&quot; &quot;MetaCardis_2020_a&quot; &quot;MetaCardis_2020_a&quot; &quot;GhensiP_2019&quot; ...
##   .. ..$ : chr  &quot;PNP_Main_553&quot; &quot;M0x20MCx1812&quot; &quot;M0x20MCx1360&quot; &quot;SP_101SPI_T016&quot; ...
##   .. ..$ : chr  &quot;PNP_Main_553&quot; &quot;M0x20MCx1812&quot; &quot;M0x20MCx1360&quot; &quot;sub_101&quot; ...
##   .. ..$ : chr  &quot;stool&quot; &quot;stool&quot; &quot;stool&quot; &quot;oralcavity&quot; ...
##   .. ..$ : chr  &quot;no&quot; &quot;no&quot; &quot;yes&quot; NA ...
##   .. ..$ : chr  &quot;control&quot; &quot;T2D&quot; &quot;IGT&quot; &quot;control&quot; ...
##   .. ..$ : chr  &quot;healthy&quot; &quot;T2D&quot; &quot;IGT;MS&quot; &quot;healthy&quot; ...
##   .. ..$ : int  51 63 51 NA 27
##   .. ..$ : chr  &quot;adult&quot; &quot;adult&quot; &quot;adult&quot; &quot;adult&quot; ...
##   .. ..$ : chr  &quot;female&quot; &quot;male&quot; &quot;female&quot; NA ...
##   .. ..$ : chr  &quot;ISR&quot; &quot;DEU&quot; &quot;DEU&quot; &quot;ITA&quot; ...
##   .. ..$ : chr  NA &quot;Leipzig&quot; &quot;Leipzig&quot; NA ...
##   .. ..$ : chr  &quot;no&quot; &quot;no&quot; &quot;no&quot; &quot;no&quot; ...
##   .. ..$ : chr  &quot;IlluminaHiSeq&quot; &quot;IonProton&quot; &quot;IonProton&quot; &quot;IlluminaHiSeq&quot; ...
##   .. ..$ : chr  NA NA NA NA ...
##   .. ..$ : int  26590418 34880489 34880489 33127901 22699609
##   .. ..$ : int  28756422 15374062 20888458 20061461 94436101
##   .. ..$ : num  2.52e+09 2.33e+09 3.20e+09 1.99e+09 8.71e+09
##   .. ..$ : int  20 NA NA 75 60
##   .. ..$ : int  80 NA NA 100 101
##   .. ..$ : chr  &quot;ERR1136663&quot; &quot;ERR4562131&quot; &quot;ERR4087400&quot; &quot;SRR9217435&quot; ...
##   .. ..$ : num  NA NA 6.78 NA NA
##   .. ..$ : chr  &quot;no&quot; &quot;antidiab;metformin;dppiv&quot; &quot;antihta;ca2_cbl&quot; NA ...
##   .. ..$ : chr  &quot;Jacob_Wirbel;Paolo_Manghi&quot; &quot;Paolo_Manghi&quot; &quot;Paolo_Manghi&quot; &quot;Paolo_Manghi&quot; ...
##   .. ..$ : chr  &quot;no&quot; NA NA NA ...
##   .. ..$ : num  23.7 27.7 50.1 NA 23
##   .. ..$ : chr  NA &quot;yes&quot; &quot;no&quot; &quot;no&quot; ...
##   .. ..$ : chr  &quot;rectal_swab&quot; NA NA &quot;subgingival_plaque&quot; ...
##   .. ..$ : chr  &quot;no_immuno_suppressive;no_T2D;no_T1D;no_related_treatments;no_psychiatric_diseases;no_gastro_intestinal_disorder;non_celiac&quot; NA NA NA ...
##   .. ..$ : num  NA NA 210 NA NA
##   .. ..$ : num  NA NA 240 NA NA
##   .. ..$ : num  NA NA 5.73 NA NA
##   .. ..$ : chr  NA NA NA &quot;yes&quot; ...
##   .. ..$ : chr  NA NA NA &quot;implant&quot; ...
##   .. ..$ : chr  NA NA NA &quot;yes&quot; ...
##   .. ..$ : int  NA NA NA 4 NA
##   .. ..$ : int  NA NA NA 3 NA
##   .. ..$ : int  NA NA NA 4 NA
##   .. ..$ : int  NA NA NA 3 NA
##   .. ..$ : int  NA NA 4 NA NA
##   .. ..$ : chr  &quot;PRJEB11532&quot; &quot;PRJEB38742&quot; &quot;PRJEB37249&quot; &quot;PRJNA547717&quot; ...
##   ..@ names    : chr  &quot;study_name&quot; &quot;sample_id&quot; &quot;subject_id&quot; &quot;body_site&quot; ...
##   ..@ row.names: chr  &quot;sa1&quot; &quot;sa2&quot; &quot;sa3&quot; &quot;sa4&quot; ...
##   ..@ .S3Class : chr &quot;data.frame&quot;</code></pre>
<p>Format the dataframe to work well with the sourmash phyloseq
object</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">names</span>(metadata) <span class="ot">&lt;-</span> <span class="fu">tolower</span>(<span class="fu">names</span>(metadata))</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a><span class="fu">rownames</span>(metadata) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>ncbi_accession</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(metadata, <span class="at">format=</span><span class="st">&#39;markdown&#39;</span>)</span></code></pre></div>
<table class="kable_wrapper">
<tbody>
<tr>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ZeeviD_2015</td>
</tr>
<tr class="even">
<td align="left">MetaCardis_2020_a</td>
</tr>
<tr class="odd">
<td align="left">MetaCardis_2020_a</td>
</tr>
<tr class="even">
<td align="left">GhensiP_2019</td>
</tr>
<tr class="odd">
<td align="left">HMP_2012</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PNP_Main_553</td>
</tr>
<tr class="even">
<td align="left">M0x20MCx1812</td>
</tr>
<tr class="odd">
<td align="left">M0x20MCx1360</td>
</tr>
<tr class="even">
<td align="left">SP_101SPI_T016</td>
</tr>
<tr class="odd">
<td align="left">SRS013723</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PNP_Main_553</td>
</tr>
<tr class="even">
<td align="left">M0x20MCx1812</td>
</tr>
<tr class="odd">
<td align="left">M0x20MCx1360</td>
</tr>
<tr class="even">
<td align="left">sub_101</td>
</tr>
<tr class="odd">
<td align="left">HMP_2012_159268001</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">stool</td>
</tr>
<tr class="even">
<td align="left">stool</td>
</tr>
<tr class="odd">
<td align="left">stool</td>
</tr>
<tr class="even">
<td align="left">oralcavity</td>
</tr>
<tr class="odd">
<td align="left">oralcavity</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">no</td>
</tr>
<tr class="even">
<td align="left">no</td>
</tr>
<tr class="odd">
<td align="left">yes</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">control</td>
</tr>
<tr class="even">
<td align="left">T2D</td>
</tr>
<tr class="odd">
<td align="left">IGT</td>
</tr>
<tr class="even">
<td align="left">control</td>
</tr>
<tr class="odd">
<td align="left">control</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">healthy</td>
</tr>
<tr class="even">
<td align="left">T2D</td>
</tr>
<tr class="odd">
<td align="left">IGT;MS</td>
</tr>
<tr class="even">
<td align="left">healthy</td>
</tr>
<tr class="odd">
<td align="left">healthy</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">51</td>
</tr>
<tr class="even">
<td align="right">63</td>
</tr>
<tr class="odd">
<td align="right">51</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">27</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">adult</td>
</tr>
<tr class="even">
<td align="left">adult</td>
</tr>
<tr class="odd">
<td align="left">adult</td>
</tr>
<tr class="even">
<td align="left">adult</td>
</tr>
<tr class="odd">
<td align="left">adult</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">female</td>
</tr>
<tr class="even">
<td align="left">male</td>
</tr>
<tr class="odd">
<td align="left">female</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">male</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ISR</td>
</tr>
<tr class="even">
<td align="left">DEU</td>
</tr>
<tr class="odd">
<td align="left">DEU</td>
</tr>
<tr class="even">
<td align="left">ITA</td>
</tr>
<tr class="odd">
<td align="left">USA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Leipzig</td>
</tr>
<tr class="odd">
<td align="left">Leipzig</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">no</td>
</tr>
<tr class="even">
<td align="left">no</td>
</tr>
<tr class="odd">
<td align="left">no</td>
</tr>
<tr class="even">
<td align="left">no</td>
</tr>
<tr class="odd">
<td align="left">no</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">IlluminaHiSeq</td>
</tr>
<tr class="even">
<td align="left">IonProton</td>
</tr>
<tr class="odd">
<td align="left">IonProton</td>
</tr>
<tr class="even">
<td align="left">IlluminaHiSeq</td>
</tr>
<tr class="odd">
<td align="left">IlluminaHiSeq</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Qiagen</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">26590418</td>
</tr>
<tr class="even">
<td align="right">34880489</td>
</tr>
<tr class="odd">
<td align="right">34880489</td>
</tr>
<tr class="even">
<td align="right">33127901</td>
</tr>
<tr class="odd">
<td align="right">22699609</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">28756422</td>
</tr>
<tr class="even">
<td align="right">15374062</td>
</tr>
<tr class="odd">
<td align="right">20888458</td>
</tr>
<tr class="even">
<td align="right">20061461</td>
</tr>
<tr class="odd">
<td align="right">94436101</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">2515847316</td>
</tr>
<tr class="even">
<td align="right">2326962418</td>
</tr>
<tr class="odd">
<td align="right">3200734644</td>
</tr>
<tr class="even">
<td align="right">1987501438</td>
</tr>
<tr class="odd">
<td align="right">8712278548</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">20</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">75</td>
</tr>
<tr class="odd">
<td align="right">60</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">80</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">100</td>
</tr>
<tr class="odd">
<td align="right">101</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ERR1136663</td>
</tr>
<tr class="even">
<td align="left">ERR4562131</td>
</tr>
<tr class="odd">
<td align="left">ERR4087400</td>
</tr>
<tr class="even">
<td align="left">SRR9217435</td>
</tr>
<tr class="odd">
<td align="left">SRR059888</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">6.78</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">no</td>
</tr>
<tr class="even">
<td align="left">antidiab;metformin;dppiv</td>
</tr>
<tr class="odd">
<td align="left">antihta;ca2_cbl</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Jacob_Wirbel;Paolo_Manghi</td>
</tr>
<tr class="even">
<td align="left">Paolo_Manghi</td>
</tr>
<tr class="odd">
<td align="left">Paolo_Manghi</td>
</tr>
<tr class="even">
<td align="left">Paolo_Manghi</td>
</tr>
<tr class="odd">
<td align="left">Paolo_Manghi</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">no</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">23.70000</td>
</tr>
<tr class="even">
<td align="right">27.67959</td>
</tr>
<tr class="odd">
<td align="right">50.14402</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">23.00000</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">yes</td>
</tr>
<tr class="odd">
<td align="left">no</td>
</tr>
<tr class="even">
<td align="left">no</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">rectal_swab</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">subgingival_plaque</td>
</tr>
<tr class="odd">
<td align="left">supragingival_plaque</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">no_immuno_suppressive;no_T2D;no_T1D;no_related_treatments;no_psychiatric_diseases;no_gastro_intestinal_disorder;non_celiac</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">210.3648</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">240.0247</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">5.73</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">yes</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">implant</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">yes</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">4</td>
</tr>
<tr class="even">
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">NA</td>
</tr>
</tbody>
</table>
</td>
<td>

<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PRJEB11532</td>
</tr>
<tr class="even">
<td align="left">PRJEB38742</td>
</tr>
<tr class="odd">
<td align="left">PRJEB37249</td>
</tr>
<tr class="even">
<td align="left">PRJNA547717</td>
</tr>
<tr class="odd">
<td align="left">PRJNA48479</td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>

<p>Finally, merge the metadata with the sourmash phyloseq object</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>pseq_final <span class="ot">&lt;-</span> <span class="fu">merge_phyloseq</span>(physeq, metadata)</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a>pseq_final</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 15 taxa and 5 samples ]
## sample_data() Sample Data:       [ 5 samples by 41 sample variables ]
## tax_table()   Taxonomy Table:    [ 15 taxa by 8 taxonomic ranks ]</code></pre>

</body>
</html>
